src/HttpClientFactory/Http/test/DependencyInjection/HttpClientFactoryServiceCollectionExtensionsTest.cs

```diff
@@ -771,6 +774,45 @@ public async Task AddHttpClient_MessageHandler_Scope_TransientDependency()
             }
         }
 
+        [Fact]
+        public void AddHttpClient_GetAwaiterAndResult_InSingleThreadedSynchronizationContext_ShouldNotHangs()
```

Comment:
I think we need a way for this test to avoid the hang if a bug is introduced. We need a timeout somewhere.